zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -113,9 +115,12 @@
     private AtomicLong observerCounter = new AtomicLong(-1);
 
     /*
-     * Protocol identifier used among peers
+     * Protocol identifier used among peers (must be a negative number for backward compatibility reasons)
      */
-    public static final long PROTOCOL_VERSION = -65535L;
+    // the following protocol version was sent in every connection initiation message since ZOOKEEPER-2186 released in 3.4.7
+    public static final long PROTOCOL_VERSION_3_4_7 = -65536L;
```

Comment:
I would call this constant PROTOCOL_VERSION_V1
and the other one PROTOCOL_VERSION_V2

Reply:
The reason why I named this way was that I am not sure if we won't need to add other previous versions later (e.g. to support rolling restart from pre-3.4.7 versions). But giving a second thought, I like your proposal better. These are just constants, we can rename them later in any ways. I will change this in the next commit.